### DISCUSSION TITLE
Buildpacks support for Azure functions with multiple entrypoints

### DISCUSSION THREAD

**Discussion by AshwinSridharan0410** [paketo-buildpacks/feedback#258] [.NET Core Team]
I have an Azure function which has multiple entrypoints, (meaning. there are a lot of run() functions associated with triggers of multiple kinds)
So, each run() function corresponds to an entrypoint.

Example:-
using System;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Host;
using Microsoft.Extensions.Logging;

namespace MyAzureFunctionsNamespace
{
    public static class MultipleFunctions
    {
        [FunctionName("TimerTriggerFunction")]
        public static void TimerTriggerFunction([TimerTrigger("0 */5 * * * *")] TimerInfo myTimer, ILogger log)
        {
            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
        }

        [FunctionName("HttpTriggerFunction")]
        public static void HttpTriggerFunction([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req, ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");

            // Your HTTP trigger logic here

            string name = req.Query["name"];

            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
            dynamic data = JsonConvert.DeserializeObject(requestBody);
            name = name ?? data?.name;

            return new OkObjectResult($"Hello, {name}");
        }

        [FunctionName("BlobTriggerFunction")]
        public static void BlobTriggerFunction([BlobTrigger("my-container/{name}", Connection = "AzureWebJobsStorage")] Stream myBlob, string name, ILogger log)
        {
            log.LogInformation($"Blob trigger function Processed blob\n Name:{name} \n Size: {myBlob.Length} Bytes");
        }
    }
}
Here there are multiple triggers(entrypoints). How will buildpacks support all of these ?

When looked upon the code of dotnet buildpacks, 
I understood this logic,
When we build upon a .csproj file, in the backend, it is looking for a runtime.config.json which gets the name of csproj file.
Say. abc.csproj generates  abc.dll and abc.runtime.config.json
And buildpacks looks for an entrypoint function with the same name runtimeconfig.json is generated.
Please correct my understanding here.
This would work fine , if I just have a single trigger function acting as the entrypoint. So I will change the name of the function to that, thus buildpacks will identify it and set the process type and start the exporting process in the lifecycle.
How will it work for multiple entrypoints as stated in the above example ?

**Comment by AshwinSridharan0410** [comment]
@andymoe @nebhale @anthonydahanne @ryanmoran 
Can you provide your views here

**Reply by RakeshSharnagate-eaton** [reply]
Hello Ashwin,

Are you able to run the single entrypoint .NET8 Azure function?
I am struggling to run the simple HTTP trigger Azure function on AKS.
